library/dokku_image.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.dokku_app import dokku_app_ensure_present
+
+DOCUMENTATION = """
+---
+module: dokku_image
+short_description: Pull Docker image and deploy app
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  image:
+    description:
+      - Docker image
+    required: True
+    default: null
+    aliases: []
+  user_name:
+    description:
+      - Git user.name for customizing the author's name
+    required: False
+    default: null
+    aliases: []
+  user_email:
+    description:
+      - Git user.email for customizing the author's email
+    required: False
+    default: null
+    aliases: []
+  build_dir:
+    description:
+      - Specifiy custom build directory for a custom build context
+    required: False
+    default: null
+    aliases: []
+author: Simo Aleksandrov
+"""
+
+EXAMPLES = """
+- name: Pull and deploy meilisearch
+  dokku_image:
+      app: meilisearch
+      image: getmeili/meilisearch:v0.24.0rc1
+- name: Pull and deploy image with custom author
+  dokku_image:
+      app: hello-world
+      user_name: Elliot Alderson
+      user_email: elliotalderson@protonmail.ch
+      repository: hello-world:latest
+- name: Pull and deploy image with custom build dir
+  dokku_image:
+      app: hello-world
+      build_dir: /path/to/build
+      repository: hello-world:latest
+"""
+
+
+def dokku_git_sha(data):
+    """Get SHA of current app repository.
+
+    Returns `None` if app does not exist.
+    """
+    command_git_report = "dokku git:report {app} --git-sha".format(app=data["app"])
+    try:
+        sha = subprocess.check_output(
+            command_git_report, stderr=subprocess.STDOUT, shell=True
+        )
+    except subprocess.CalledProcessError:
+        sha = None
+
+    return sha
+
+
+def dokku_image(data):
+    # create app (if not exists)
+    is_error, has_changed, meta = dokku_app_ensure_present(data)
+    meta["present"] = False  # meaning: requested *version* of app is present
+    if is_error:
+        return (is_error, has_changed, meta)
+
+    sha_old = dokku_git_sha(data)
+
+    # get image
+    command_git_from_image = "dokku git:from-image {app} {image}".format(
+        app=data["app"], image=data["image"]
+    )
+    if data["user_name"]:
+        command_git_from_image += " {user_name}".format(user_name=data["user_name"])
+    if data["user_email"]:
+        command_git_from_image += " {user_email}".format(user_email=data["user_email"])
+    if data["build_dir"]:
+        command_git_from_image += " --build-dir {build_dir}".format(
```

Comment:
perhaps put quotes around the directory in case it contains spaces / ...

you could also check whether the directory exists but I guess dokku will check that as well, so no necessary